_posts/en/newsletters/2024-05-17-newsletter.md

```diff
@@ -0,0 +1,185 @@
+  contract protocol can prevent any one (or more) of the parties from
+  withdrawing money from the contract unless that party successfully
+  executes an arbitrary program specified by the contract.
+
+  Lerner links to a [paper][bitvmx paper] about BitVMX which compares it
+  to the original BitVM (see [Newsletter #273][news273 bitvm]) and to
+  the limited details available about follow-up projects from the
+  original BitVM developers.  An accompanying [website][bitvmx website]
+  provides additional information in a slightly less technical form.
+
+- **Continued discussion about updating BIP2:** Mark "Murch" Erhardt
+  [continued][erhardt bip2] the discussion on the Bitcoin-Dev mailing
+  list about updating [BIP2][], which is the document that currently
+  describes the Bitcoin improvement proposals (BIP) process.  His email
+  describes several problems, suggests solutions for many of them,
+  and solicits feedback on his suggestions as well as proposals for
+  solutions to the remaining problems.  For previous discussion about
+  updating BIP2, see [Newsletter #297][news297 bip2]. <!-- TODO
+  (optional): harding to add topic page for something like
+  "Specification management" to track BIP and BINANA meta discussion -->
+
+## Releases and release candidates
+
+*New releases and release candidates for popular Bitcoin infrastructure
+projects.  Please consider upgrading to new releases or helping to test
+release candidates.*
+
+- [LND v0.18.0-beta.rc2][] is a release candidate for the next major
+  version of this popular LN node implementation.
+
+## Notable code and documentation changes
+
+_Notable recent changes in [Bitcoin Core][bitcoin core repo], [Core
+Lightning][core lightning repo], [Eclair][eclair repo], [LDK][ldk repo],
+[LND][lnd repo], [libsecp256k1][libsecp256k1 repo], [Hardware Wallet
+Interface (HWI)][hwi repo], [Rust Bitcoin][rust bitcoin repo], [BTCPay
+Server][btcpay server repo], [BDK][bdk repo], [Bitcoin Improvement
+Proposals (BIPs)][bips repo], [Lightning BOLTs][bolts repo],
+[Bitcoin Inquisition][bitcoin inquisition repo], and [BINANAs][binana
+repo]._
+
+- [Core Lightning #7190][] adds an additional offset `chainlag` into the CLTV
+  value calculation to correctly target the current block height instead of the
+  sync height. Payments are now safe and allowed during the blockchain sync
+  process.
+
+- [LDK #2973][] implements support for `OnionMessenger` to intercept messages on
+  behalf of offline peers. It generates events on message interception and on
+  the peer's return to online status for forwarding. Users should maintain an
+  allow list to only store messages for relevant peers. This is a stepping stone
+  to supporting offline payments through `held_htlc_available` [BOLTs
+  #989][], which would allow an LSP (Lightning Service Provider) to hold onto
+  an [HTLC][topic htlc] until a user comes back online.
+
+- [LDK #2907][] extends `OnionMessage` handling to accept an optional
```

Comment:
@Gustavojfe just a note that I'm going to update this description.  I infer from your description that Alice sends an HTLC to Bob, Bob realizes Carol is offline and so holds the message until she comes back online, then Bob delivers the HTLC.  Wallets have tried that before, but it's problematic because not immediately settling the HTLC means every hop along the path has their liquidity reduced.  That looks like an attack.

What I believe this PR is actually describing is that Alice wants to pay Carol but doesn't know if Carol is online, so Alice sends an _onion message_ to Carol via Bob (an onion message is data only, no money); Bob realizes Carol is offline and so holds the message until she comes back online; then Bob delivers the onion message to Carol; Carol opens the onion message which says "Alice wants to pay you, reply when you're ready to accept the payment"; Carol replies and Alice sends the payment.